test/unit_tests/autoqasm/test_api.py

```diff
@@ -1266,3 +1266,39 @@ def test(int[32] a, int[32] b) {
 test(2, 3);
 test(4, 5);"""
     assert main.build().to_ir() == expected
+
+
+class TestTypecasting:
+    def test_int_typecasting_on_measure(self):
+        @aq.main(num_qubits=2)
+        def main():
+            test = int(measure([0, 1]))
+
+        expected_ir = """OPENQASM 3.0;
+qubit[2] __qubits__;
+bit[2] __bit_0__ = "00";
+__bit_0__[0] = measure __qubits__[0];
+__bit_0__[1] = measure __qubits__[1];
+int[32] test = __bit_0__;"""
+        assert main.build().to_ir() == expected_ir
+
+    def test_int_typecasting_on_string(self):
+        @aq.main(num_qubits=2)
+        def main():
+            test = int("101", 2)
+
+        expected_ir = """OPENQASM 3.0;
+qubit[2] __qubits__;"""
+        assert main.build().to_ir() == expected_ir
+
+    def test_nested_int_typecasting(self):
+        @aq.main(num_qubits=2)
+        def main():
+            test = 2 * int(measure([0, 1]))
+
+        expected_ir = """OPENQASM 3.0;
+qubit[2] __qubits__;
+bit[2] __bit_0__ = "00";
+__bit_0__[0] = measure __qubits__[0];
+__bit_0__[1] = measure __qubits__[1];"""
```

Comment:
This doesn't look right. I would expect the output IR to contain something like:
```
int[32] __int_1__ = __bit_0__;
int[32] test = 2 * __int_1__;
```

Reply:
For debugging, might be interesting to know what the IR output is if you modify the program to be like this:
```
        @aq.main(num_qubits=2)
        def main():
            test = int(2 * measure([0, 1]))
```
or like this:
```
        @aq.main(num_qubits=2)
        def main():
            test = 2 * int(measure([0, 1]))
            return test
```

Reply:
I checked and using
```
@aq.main(num_qubits=2)
        def main():
            test = 2 * IntVar(measure([0, 1]))
```
results in the same IR. I think the discrepancy came from the fact that the variable `test` was not being used after declared. Your second suggestion helped me realize that adding `return test` or other lines that make use of `test` results in an IR with the int[32] lines. I update the unit test to reflect this.

On a second note, I did not add support for arithmetic operations on BitVara in this PR because I focused on int typecasting. To add the arithmetic operations, would you suggest I add methods like `__mul__` and `__rmul__` to the `BitVar` class (where `self` gets casted to an `IntVar`) or use a converter to intercept the arithmetic operations and cast the BitVars to IntVars?